src/netex_idf/lines.rs

```diff
@@ -56,12 +54,19 @@ fn load_netex_lines(
             for line in lines_node.children().filter(|e| e.name() == "Line") {
                 let id = line.try_attribute("id")?;
                 let name = line.try_only_child("Name")?.text().parse()?;
-                let code = line.only_child("ShortName").map(Element::text);
+                let code = line
+                    .try_only_child_with_filter("PublicCode", |e| !e.text().is_empty())
+                    .or_else(|_| line.try_only_child("ShortName"))
+                    .map(Element::text)
+                    .ok();
                 let private_code = line.only_child("PrivateCode").map(Element::text);
-                let network_id = if let Some(network_id) = map_line_network.get(&id) {
-                    network_id.to_string()
+                let network_id: String = line
+                    .try_only_child("RepresentedByGroupRef")?
+                    .try_attribute("ref")?;
+                let network_id = if let Some(_network) = networks.get(&network_id) {
+                    network_id
                 } else {
-                    warn!("Failed to find network for line {}", id);
+                    warn!("Failed to find network {} for line {}", network_id, id);
                     continue;
                 };
```

Comment:
I think you can do something simpler like this.
```
if networks.get(&network_id).is_none() {
    warn!("Failed to find network {} for line {}", network_id, id);
    continue;
}
```

Reply:
same comment